### PR TITLE
Initial template tool

### PR DESCRIPTION
**Pull request by aerkiaga**
Developed by @ghutchis, this PR finishes a basic implementation with monodentate ligands only.

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

File: avogadro/qtplugins/templatetool/templatetoolwidget.cpp [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+}
+
+void TemplateToolWidget::selectElement(unsigned char element)
+{
+  int curIndex = element > 0 ? m_ui->elementComboBox->findData(element) : -1;
+  if (curIndex >= 0)
+    m_ui->elementComboBox->setCurrentIndex(curIndex);
+  else {
+    addUserElement(element);
+    int curIndex = m_ui->elementComboBox->findData(element);
```
[the remark this answers is not in the document]
Comment (ghutchis) [reply avogadro/qtplugins/templatetool/templatetoolwidget.cpp:210]: This should definitely be fixed up...

**Comment by ghutchis** [comment]
There are a bunch of minor things, but this is a good start. (I'm thinking it's better to put ligands, etc. in a separate rep, but that's not urgent.)

**Comment by ghutchis** [comment]
One thing that *does* need to get fixed is that the attachment point for aqua ligands is in the plane of the H-O-H atoms (e.g., https://www.researchgate.net/figure/Structures-of-hexa-aqua-metal-complexes-a-Protonated-state-b-Deprotonated-state-Magenta_fig3_342985905)

**Comment by ghutchis** [comment]
The other thing is that I'd suggest something more like `InsertFragment::performInsert` instead of adding atoms one at a time. That way the undo/redo stack is one change. You can read in the template or center molecule, translate / rotate the coordinates and then call something like:

```
m_molecule->appendMolecule(newMol, tr("Insert Template"));
```

**Comment by aerkiaga** [comment]
I've solved most of the issues. Still, `appendMolecule` is only used for the template center, not for ligands. I suppose I can just perform the same process, then create a bond between the appropriate atoms as a separate step?

**Comment by aerkiaga** [comment]
Kinda working now. It's now three undo steps: delete hydrogen, insert fragment, add bond.

**Comment by ghutchis** [comment]
Okay, I'm willing to merge this and we can work from here. Is Slack still a good way to chat? I have a few ideas on bidentate ligands, etc.

**Comment by ghutchis** [comment]
Closed by #1075

**Pull request closed without merging**